code/modules/organs/organ.dm

```diff
@@ -140,6 +144,32 @@ var/list/organ_cache = list()
 	if(damage >= max_damage)
 		die()
 
+/obj/item/organ/proc/is_preserved()
+	if(istype(loc,/obj/item/device/mmi))
+		germ_level = max(0, germ_level - 1) // So a brain can slowly recover from being left out of an MMI
+		return 1
+	if(is_found_within(/obj/item/bodybag/cryobag))
+		return 1
+	if(is_found_within(/obj/structure/closet/crate/freezer))
+		return 1
+	if(istype(loc,/turf))
+		if(needs_preservation_update)
+			// I don't want to loop through everything in the tile constantly, especially since it'll be a pile of organs
+			// if the virologist releases gibbingtons again or something
+			// There's probably a much less silly way of doing this, but BYOND native algorithms are stupidly naive
+			is_in_freezer = 0
+			for(var/obj/structure/closet/crate/freezer/F in loc.contents)
+				if(F.opened)
+					is_in_freezer = 1 // on the same tile, close enough, should keep organs much fresher on avg
+					break
+			needs_preservation_update = 0
+			spawn(100) // (100 / 10) once every 10 seconds
+				needs_preservation_update = 1
```

Comment:
Hrmmm, this should only fire every 10 seconds, is this alright for a spawn() in process()?